src/components/Select/tests/Select.test.tsx

```diff
@@ -1,64 +1,72 @@
 import React from 'react';
 import {InlineError, Icon, Labelled} from 'components';
-// eslint-disable-next-line no-restricted-imports
-import {mountWithAppProvider, ReactWrapper} from 'test-utilities/legacy';
+import {mountWithApp} from 'test-utilities';
 import {CircleTickOutlineMinor} from '@shopify/polaris-icons';
+import {act} from 'react-dom/test-utils';
 
 import {Select} from '../Select';
 
 describe('<Select />', () => {
   describe('onChange()', () => {
     it('is called with the value of the newly-selected option', () => {
       const spy = jest.fn();
-      const element = mountWithAppProvider(
+      const element = mountWithApp(
         <Select
           id="MySelect"
           label="Select"
           options={['one', 'two']}
           onChange={spy}
         />,
       );
-      (element.find('select') as any).instance().value = 'two';
-      element.find('select').simulate('change');
+
+      const select = element.find('select')! as any;
+      act(() => {
+        select.value = 'two';
+      });
+
+      const event = {
+        currentTarget: select,
+      };
+
+      select.trigger('onChange', event);
       expect(spy).toHaveBeenCalledWith('two', 'MySelect');
     });
   });
 
   describe('onFocus()', () => {
     it('is called when the select is focused', () => {
       const spy = jest.fn();
-      mountWithAppProvider(
+      mountWithApp(
         <Select label="Select" options={[]} onFocus={spy} onChange={noop} />,
       )
-        .find('select')
-        .simulate('focus');
+        .find('select')!
+        .trigger('onFocus');
       expect(spy).toHaveBeenCalled();
     });
   });
 
   describe('onBlur()', () => {
     it('is called when the select is blurred', () => {
       const spy = jest.fn();
-      const element = mountWithAppProvider(
+      const element = mountWithApp(
         <Select label="Select" options={[]} onBlur={spy} onChange={noop} />,
       );
-      element.find('select').simulate('focus');
```

Comment:
Here is not necessary to call onFocus?

Reply:
I don't think it's needed to focus on the event as we're just testing the callback when onBlur is triggered.